Add Expire() function

morses-code opened the pull request:
Make sure tests pass, review implementation of Expire() function, check how Expire() is called within GetByID()

morses-code committed:
Added DeleteAll func with tests

- Created `DeleteAll()` function
- Added redis implementation for `FlushAll()`
- Added tests for `DeleteAll()`
- Added additional parameter to `setUpMocks()` and amended all calls
- Added `FlushAll()` to `RedisClienter` interface
- Generated new mock file

morses-code committed:
Update LastAccessed when Session is accessed by GetByID()

- Update sessions LastAccessed when `GetByID()` is called
- Add redis `Expire()` implementation
- Added new test to check LastAccessed is updated
- Added new test when `Expire()` returns an error
- Added parameter to `setUpMocks()`
- Added constant for LastAccessed on sample data
- Added `Expire()` to `RedisClienter` interface
- Generated updated mock file

DavidSubiros commented on client.go:
```diff
@@ -93,9 +93,25 @@ func (c *Client) GetByID(id string) (*session.Session, error) {
 	return s, nil
 }
 
+func (c *Client) DeleteAll() error {
```
I think this was built on top of the other PR https://github.com/ONSdigital/dp-redis/pull/6
When @CarlHembrough comment is resolved (either making a change or not), I will resolve this comment too.

DavidSubiros commented on client_test.go:
```diff
@@ -152,23 +169,56 @@ func TestClient_Set(t *testing.T) {
 
 			Convey("Then redis client.Get is called and returns the session", func() {
 				So(mockRedisClient.GetCalls(), ShouldHaveLength, 1)
+				So(mockRedisClient.ExpireCalls(), ShouldHaveLength, 1)
```
Maybe it would be worth validating that the TTL was extended by the intended duration?
You could add the following extra check:
```
So(mockRedisClient.ExpireCalls()[0].Expiration, ShouldEqual, testTTL)
```

morses-code replied:
Thanks David, I have now added this check. I was trying to figure out how to do this, so thank you 👍

DavidSubiros reviewed (commented):
Everything looks good to me. Will approve as soon as the other PR is approved. Also suggested an extra check in a unit test test.

morses-code committed:
Added check that TTL is expiration

- Updated `GetByID()` test to check that TTL is refreshed by the expected value
